packages/block-library/src/post-time-to-read/edit.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { _x, _n, __, sprintf } from '@wordpress/i18n';
+import { useMemo } from '@wordpress/element';
+import {
+	AlignmentControl,
+	BlockControls,
+	useBlockProps,
+} from '@wordpress/block-editor';
+import { __unstableSerializeAndClean } from '@wordpress/blocks';
+import { useEntityProp, useEntityBlockEditor } from '@wordpress/core-data';
+import { count as wordCount } from '@wordpress/wordcount';
+
+/**
+ * Average reading rate - based on average taken from
+ * https://irisreading.com/average-reading-speed-in-various-languages/
+ * (Characters/minute used for Chinese rather than words).
+ */
+const AVERAGE_READING_RATE = 189;
+
+function PostTimeToReadEdit( { attributes, setAttributes, context } ) {
+	const { textAlign } = attributes;
+	const { postId, postType } = context;
+
+	const [ contentStructure ] = useEntityProp(
+		'postType',
+		postType,
+		'content',
+		postId
+	);
+
+	const [ blocks ] = useEntityBlockEditor( 'postType', postType, {
+		id: postId,
+	} );
+
+	const minutesToReadString = useMemo( () => {
+		// Replicates the logic found in getEditedPostContent().
+		let content;
+		if ( contentStructure instanceof Function ) {
+			content = contentStructure( { blocks } );
+		} else if ( blocks ) {
+			// If we have parsed blocks already, they should be our source of truth.
+			// Parsing applies block deprecations and legacy block conversions that
+			// unparsed content will not have.
+			content = __unstableSerializeAndClean( blocks );
+		} else {
+			content = contentStructure;
+		}
+
+		/*
+		 * translators: If your word count is based on single characters (e.g. East Asian characters),
+		 * enter 'characters_excluding_spaces' or 'characters_including_spaces'. Otherwise, enter 'words'.
+		 * Do not translate into your own language.
+		 */
+		const wordCountType = _x(
+			'words',
+			'Word count type. Do not translate!'
+		);
+
+		const minutesToRead = Math.round(
+			wordCount( content, wordCountType ) / AVERAGE_READING_RATE
+		);
+
+		if ( minutesToRead !== 0 ) {
+			return sprintf(
+				/* translators: %d is the number of minutes the post will take to read. */
+				_n(
+					'You can read this post in %d minute.',
+					'You can read this post in %d minutes.',
+					minutesToRead
+				),
```

Comment:
Thanks for all your works so far on this @t-hamano, it feels very close to ready now.

One concern I have is still around getting this text right. If the block is released in WordPress with this text it becomes hard to change in the future it because of backwards compatibility requirements.

The prefix / postfix option was tried, but based on feedback that doesn't seem preferable right now. If there was a desire to add it again in the future it'd be hard to do with the existing 'You can read this post in' text.

It'd be good to get your thoughts on this. I'm wondering if the text should be reduced down to just '%d minute' / '%d minutes'. That way users can mix it up with an icon (through an icon block from a plugin, core doesn't have one 😄 ), or another paragraph block, or even both.

Reply:
I agree with the last idea: it will make this block more flexible and extensible, probably increasing its use by many more people.

Reply:
Thank you for the review, @talldan, @javierarce !

>  I'm wondering if the text should be reduced down to just '%d minute' / '%d minutes'. 

I think this is a very good idea! If we include "less than 1 minute" as a variation, it may not be possible to standardize the text before and after in all languages. Furthermore, the question of whether to capitalize the first letter of "less than" or lowercase arises.

As I mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/43403#discussion_r991365917), I am concerned about the separation of paragraphs, but the freedom of variation would be convenient for users.

Reply:
Agreed, we should bring this down to just displaying the time and no extra wording around it.

Reply:
> I am concerned about the separation of paragraphs

Agree with that, and it came to my mind too. It wouldn't really be semantically correct to output paragraphs in a row like this, when the user is essentially trying to concatenate text. Screenreaders may also not announce the text properly.

I think it should be considered a separate problem to solve. There are likely more use cases beyond this block. I don't know if the dynamic tokens system is considered a solution to this, while I read the initial proposal, it sounds like the outcome has changed quite a bit.

There are other solutions too, like a variation of a paragraph that outputs a span, but that kind of thing would require users knowing about html.